### PR TITLE
Add circleci.yaml for tests

### PR DESCRIPTION
**Pull request by kerin**
also:

- output pytest results in junit.xml format
- comments out example tests from template project
- use `STATICFILES_STORAGE = "whitenoise.storage.CompressedManifestStaticFilesStorage"` in prod env only
- fix template test errors in default index
- only initialise Sentry when `SENTRY_DSN` env var is set
- drop code coverage threshold from 100% to 70
- remove humans.txt test

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -74,3 +74,4 @@ flake8-isort = "^2.8.0"
 black = {version = "^19.10b0", allow-prereleases = true}
 pylint = "^2.4.4"
 autopep8 = "^1.4.4"
+polint = "^0.4"
```
Comment (r4vi) [inline pyproject.toml]: remove?

File: docker/ci.sh [hunk 2]
```diff
@@ -66,10 +66,10 @@ run_ci () {
   yamllint -d '{"extends": "default", "ignore": ".venv"}' -s .
 
   # Checking `.env` files:
-  dotenv-linter config/.env config/.env.template
+  # dotenv-linter config/.env config/.env.template
```
Comment (r4vi) [inline docker/ci.sh]: just delete these checks?

File: .circleci/config.yml [hunk 3]
```diff
@@ -61,8 +61,9 @@ jobs:
       - run:
           name: Run pytest
           command: |
-            poetry run pytest --dead-fixtures --dup-fixtures &&
-            poetry run pytest
+            poetry run pytest --dead-fixtures --dup-fixtures \
```
Comment (r4vi) [inline .circleci/config.yml]: I think you can tell circle to store these things are artifacts. Would be good to save coverage and junit so we can look at them from the circle UI for each build

File: docker/ci.sh (hunk 2 above)
Comment (kerin) [reply docker/ci.sh]: 👍

File: pyproject.toml (hunk 1 above)
Comment (kerin) [reply pyproject.toml]: binned

File: .circleci/config.yml (hunk 3 above)
Comment (kerin) [reply .circleci/config.yml]: https://app.circleci.com/jobs/github/uktrade/legal-basis-api/49/artifacts

**Pull request merged**